api/src/main/java/org/apache/iceberg/exceptions/FileIODeletionException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.exceptions;
+
+import java.util.Collection;
+
+public class FileIODeletionException extends RuntimeException {
+  public FileIODeletionException(Collection<String> failedDeletions) {
```

Comment:
How about `BulkDeleteFailureException`? I don't see a need to scope this to `FileIO`.

Reply:
Originally I was thinking it makes more sense to be more generic because deleteFiles does not impose that the underlying implementation performs bulk removals against the underlying storage. But on second thought, I do agree that scoping something to FileIO for the first implementation of this API may be unnecessarily broad. Will update to use BulkDeleteFailureException.